### ISSUE TITLE
Pid for OffAngle

### ISSUE THREAD

**Issue by Penguin-4904** [RoboticsTeam4904/2016-Code#43]
We would need to switch the PIDSource in the PIDChassisController and the algorithm, on the press of a button when we want to align with the goal.

Possible solution:
Make another Chassis controller not in standard (we do not want to do this in standard) and implement the source switching there, and the algorithm.

We could switch the controller for the teleop command 

need better solutions

**Comment by carturn** [comment]
My current thoughts are:
Make teleopCommand null.
Have ChassisMove run while some button is released.
While the button is pressed run an alternate ChassisMove with an alternate PIDChassisController
The camera stuff can implement PIDSource pretty easily.

**Comment by c0d3rman** [comment]
This is a super ugly way to do this, though. We need a better way of controlling a Chassis.

**Comment by carturn** [comment]
We might need a better way of switching. However, I am actually fairly certain that we should be changing the running command when doing autoalign type tasks. That will make handling interrupting cases (e.g. stop autoaligning) much easier. Maybe having an always running teleopCommand is incorrect.

**Comment by c0d3rman** [comment]
I agree we should. I am not sure how to guarantee that ChassisMove will continue running after autoalign is over (as the default command is Idle). I feel like the only good place to do that would be within Chassis, but I don't want to maintain a pseudo-default.
Also, I think we should definitely move away from teleopCommand - either CommandRobotBase should provide no robot state handling (and instead let the commands handle it) or it should allow as many commands as you want. I think we need ArrayLists for teleopCommand and autonomousCommand (much like the choosers.)

**Comment by carturn** [comment]
ChassisMove can be an onlyWhileReleased, align can be an onlyWhileHeld on the same button.

I think just eliminate teleopCommand. We should leave autonomousCommand (which is selected through a chooser) because that will not change while the robot is running (if there is a need to change, there should be a CommandGroup). There definitely should not be an ArrayList of either teleopCommands or autonomousCommands. Transitions do not have a good default definition. If you need to change a command, you should either be using a command group or have the transition be button triggered. Also, we do not have ArrayLists of choosers.
